### PR TITLE
Add endpoints to set and get log levels

### PR DESCRIPTION
**Pull request by saweber**
Signed-off-by: saweber <saweber@gmail.com>

## What does this PR change?
* addition of GET /logs/level and POST /logs/level endpoints

## Does this PR relate to any other PRs?
* no

## How will this PR impact users?
* ability to get and set log levels without restarting application

## Does this PR address any GitHub or Zendesk issues?
* Closes issue in private repo

## How was this PR tested?
* manually

## Does this PR require changes to documentation?
* yes - addition of GET /logs/level and POST /logs/level endpoints

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
* I can't.

### CODE REVIEW

File: pkg/costmodel/router.go [hunk 1]
```diff
@@ -1398,6 +1398,38 @@ func (a *Accesses) Status(w http.ResponseWriter, r *http.Request, _ httprouter.P
...
+	if err != nil {
+		http.Error(w, fmt.Sprintf("unable to retrive log level"), http.StatusInternalServerError)
+		return
+	}
+	w.Write(body)
+}
+
+func (a *Accesses) SetLogLevel(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	qp := httputil.NewQueryParams(r.URL.Query())
+	level := qp.Get("level", "")
```
Comment (michaelmdresser) [inline pkg/costmodel/router.go]: If this is a POST request, perhaps we should get the data from the body? Frankly, I'm not sure if that's the right convention as I'm not an HTTP expert.

File: pkg/costmodel/router.go [hunk 2]
```diff
@@ -1398,6 +1398,38 @@ func (a *Accesses) Status(w http.ResponseWriter, r *http.Request, _ httprouter.P
...
+	err := log.SetLogLevel(level)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("level must be a valid log level according to zerolog; level given: %s", level), http.StatusBadRequest)
+		return
+	}
```
Comment (michaelmdresser) [inline pkg/costmodel/router.go]: Great error message. Would you also include `err` somewhere in the message, just in case its something unexpected?

**Review by michaelmdresser — APPROVED** [review]
This is awesome, thanks for picking this up and nice encapsulation of zerolog-specific calls in `pkg/log`!

File: pkg/costmodel/router.go (hunk 1 above)
Comment (saweber) [reply pkg/costmodel/router.go]: Hopefully we don't go too deep down the REST rabbit hole here... :smile: 

1) Since this endpoint is idempotent, the 'more correct' verb is PUT. I chose to stick with POST for consistency with other endpoints in OpenCost, as well as closed source. I'm open to making this `PUT`. I just did not want this to be the only `PUT` endpoint.
2) After investigating and refreshing on REST, using the body is not required, but is generally the better practice. Also after looking into the closed source endpoints, those consistently use the body for data instead of a query parameter.

TL;DR -> Yes, I'll make the change.

Comment (michaelmdresser) [reply pkg/costmodel/router.go]: Love it, thanks for the detail. I'm down with whatever you think is best.

**Comment by nikovacevic** [comment]
Closes https://github.com/kubecost/kubecost-cost-model/issues/932

**Review by nikovacevic — APPROVED** [review]

**Comment by nikovacevic** [comment]
@saweber I'll be happy to merge this as soon as the merge conflicts are resolved. Perhaps you can rebase tomorrow, and then let me know so I can merge?

**Comment by saweber** [comment]
@nikovacevic Done!
